elide-async/src/main/java/com/yahoo/elide/async/operation/TableExportOperation.java

```diff
@@ -0,0 +1,201 @@
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.Date;
+import java.util.UUID;
+
+import java.util.concurrent.Callable;
+
+/**
+ * TableExport Execute Operation Interface.
+ */
+@Slf4j
+public abstract class TableExportOperation implements Callable<AsyncAPIResult> {
+    private TableExportFormatter formatter;
+    @Getter private AsyncExecutorService service;
+    private Integer recordNumber = 0;
+    private TableExport exportObj;
+    private RequestScope scope;
+
+    public TableExportOperation(TableExportFormatter formatter, AsyncExecutorService service,
+            AsyncAPI exportObj, RequestScope scope) {
+        this.formatter = formatter;
+        this.service = service;
+        this.exportObj = (TableExport) exportObj;
+        this.scope = scope;
+    }
+
+    @Override
+    public AsyncAPIResult call() {
+        String apiVersion = scope.getApiVersion();
+        log.debug("TableExport Object from request: {}", exportObj);
+
+        TableExportResult exportResult = new TableExportResult();
+        try {
+            EntityProjection projection = getProjection(exportObj, apiVersion);
+
+            Observable<PersistentResource> observableResults = export(exportObj, scope, projection);
+
+            Observable<String> results = Observable.empty();
+            String preResult = formatter.preFormat(projection, exportObj);
+            results = observableResults.map(resource -> {
+                this.recordNumber++;
+                return formatter.format(resource, recordNumber);
+            });
+            String postResult = formatter.postFormat(projection, exportObj);
+
+            // Stitch together Pre-Formatted, Formatted, Post-Formatted results of Formatter in single observable.
+            Observable<String> interimResults = concatStringWithObservable(preResult, results, true);
+            Observable<String> finalResults = concatStringWithObservable(postResult, interimResults, false);
+
+            storeResults(exportObj, service.getResultStorageEngine(), finalResults);
+
+            exportResult.setUrl(new URL(generateDownloadURL(exportObj, (RequestScope) scope)));
+            exportResult.setRecordCount(recordNumber);
+        } catch (BadRequestException e) {
+            exportResult.setMessage("Download url generation failure.");
+        } catch (MalformedURLException e) {
+            exportResult.setMessage("EntityProjection generation failure.");
+        }  catch (Exception e) {
+            exportResult.setMessage(e.getMessage());
+        } finally {
+            // Follows same flow as GraphQL. The query may result in failure but request was successfully processed.
+            exportResult.setHttpStatus(200);
+            exportResult.setCompletedOn(new Date());
+            return exportResult;
+        }
+    }
+
+    private Observable<String> concatStringWithObservable(String toConcat, Observable<String> observable,
```

Comment:
Observable can not be created on a null object. Can not concat empty Observable with another observable. Hence had to rely on `boolean stringFirst` to determine the order to concat.